spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -295,19 +310,15 @@ public OutputStream getOutputStream() throws IOException {
 			throw new IllegalStateException(
 					"Cannot open an output stream to a bucket: '" + getURI() + "'");
 		}
-		else {
-			Blob blob = getBlob();
 
-			if (blob == null || !blob.exists()) {
-				if (!this.autoCreateFiles) {
-					throw new FileNotFoundException("The blob was not found: " + getURI());
-				}
-
-				blob = createBlob();
-			}
+		Blob blob = getBlob();
 
-			return Channels.newOutputStream(blob.writer());
+		if ((blob == null || !blob.exists()) && !this.autoCreateFiles) {
```

Comment:
Isn't this the same as `!isWritable()`?

Reply:
Yes, although exists() also checks whether it's a bucket, and we know it's not by now.
Also, blob.exists() is not even necessary, since it's just a `getBlob() != null` check, but I am going to keep it for now for consistency with old code. We can simplify the code in the new year.